Gems/Atom/Feature/Common/Code/Source/PostProcessing/Fsr2TaaUpscalePass.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <Atom/RHI/ScopeProducer.h>
+#include <Atom/RPI.Public/Pass/Pass.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <ffx_fsr2.h>
+
+namespace AZ::Render
+{
+    struct Fsr2TaaUpscalePassData : public RPI::PassData
+    {
+        AZ_RTTI(Fsr2TaaUpscalePassData, "{68B8A0CD-DB11-47F2-BC78-4E730C65EA73}", RPI::PassData);
+        AZ_CLASS_ALLOCATOR(Fsr2TaaUpscalePassData, SystemAllocator, 0);
+
+        static void Reflect(ReflectContext* context);
+
+        ~Fsr2TaaUpscalePassData() override = default;
+
+        bool m_depthInverted = true;
+        bool m_depthInfinite = true;
+        bool m_autoExposure = true;
+        bool m_dynamicResolution = true;
+    };
+
+    class Fsr2TaaUpscalePass
+        : public RPI::Pass
+        , public RHI::ScopeProducer
```

Comment:
I think you can use RenderPass to replace these two.

Reply:
That's true but I didn't want to use `RenderPass` since it appears to do a fair bit more than I want (in particular, SRG binding/management) and this technically isn't a "render pass."